packages/react-router/src/link.tsx

```diff
@@ -557,23 +557,24 @@ type LinkCurrentTargetElement = {
 const preloadWarning = 'Error preloading route! ☝️'
 
 export function useLinkProps<
+  TComp extends React.ElementType = 'a',
```

Comment:
Should this be `HTMLAnchorElement` by default? (see [related Discord thread](https://discord.com/channels/719702312431386674/1270774234444202005))